src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2565,6 +2570,7 @@ void Session::suspend(const SessionErrorInfo& info)
     // Notify the application of the suspension of the session if the session is
     // still in the Active state
     if (m_state == Active) {
+        call_debug_hook(SyncClientHookEvent::SessionSuspended, info);
```

Comment:
shouldn't this also be called from Session::activate in case of failure?

Reply:
I'd probably just add a new hook event for that if we needed it.